pkg/pillar/zedcloud/authen.go

```diff
@@ -62,25 +57,25 @@ func verifyAuthentication(c []byte, skipVerify bool) ([]byte, types.SenderResult
 			return nil, types.SenderStatusHashSizeError, err
 		}
 
-		cert, err := getCloudCert()
+		err := getCloudCert(ctx)
```

Comment:
It would be easier to understand if the check
if ctx.serverLeafCert == nil
was here and not in the function; if I don't already have it, then get (aka "read" or "load") the serverLeafCert.

Also, the names don't line up:
 function getCloudCert
 sets serverLeafCert and sets cloudCertHash
I think all three should have similar names.

And perhaps calling it signingCert instead of serverCert? I guess we have our own signingCert.
So serverSigningCert would make sense for all three.